### ISSUE TITLE
[Question] Is infinite zoom in the scope of this project?

### ISSUE THREAD

**Issue by Derpysphere** [Apostolique/Mitten#14]
It would take some work, but I think this project would blow up, also I kinda want infinite zoom.

**Comment by Apostolique** [comment]
Yeah, that would be one of my biggest wishes for making the app feel truly boundless, I started thinking about it in the past weeks.

I think I'll need a setup where I divide the zoom space into zones on the exponential scale. Each zone would have it's own camera and spatial partitioning. This works well when zooming out since I can just not draw lines that are too small. I'm not yet sure how I'd handle lines that are too big. All the levels above would compound which isn't that efficient and probably have floating point issues for drawing.

Moving on the horizontal or vertical axis very far could be done similarly.

I'll have to experiment to see if this idea truly works.

Of course I'd want this new feature to be backward compatible with the old canvas saves but I don't think that will be a problem.

**Comment by Derpysphere** [comment]
> Yeah, that would be one of my biggest wishes for making the app feel truly boundless, I started thinking about it in the past weeks.
> 
> I think I'll need a setup where I divide the zoom space into zones on the exponential scale. Each zone would have it's own camera and spatial partitioning. This works well when zooming out since I can just not draw lines that are too small. I'm not yet sure how I'd handle lines that are too big. All the levels above would compound which isn't that efficient and probably have floating point issues for drawing.
I think there will be some problem solving that goes into it lol, but I think your going to have to cheat a little.
> Of course I'd want this new feature to be backward compatible with the old canvas saves but I don't think that will be a problem.
I wouldn't really recommend putting any real work into making it backwards compatible, to make it infinite you're going to have to make some really big changes, but if its easy, go ahead.

I think it would be awesome.

**Comment by Apostolique** [comment]
Hey, it's already been more than one year since this issue was opened. I tried implementing this during the year but the algorithm I was going for was a bit too ambitious and I wasn't satisfied with it. I'll have to toy with other ideas. I tried other infinite canvas apps but they had issues in areas that I was expecting. Anyway, I still want this feature to happen and my silence doesn't meant that I've given up on it.

**Comment by MountainLabz** [comment]
Wow, that's cool too hear. It's awesome to see you still working on it!

**Comment by Apostolique** [comment]
I just did a release which adds true infinity. Definitely test it and report bugs if you find any.

**Comment by MountainLabz** [comment]
Cool! I'll check it out.
